### ISSUE TITLE
Feature request: Lint for deprecated namespace

### ISSUE THREAD

**Issue by serioga** [clj-kondo/clj-kondo#1311]
**Is your feature request related to a problem? Please describe.**

Namespace declaration allows `:deprecated` field in metadata.
Cursive marks all usages of the namespace as deprecated (see screenshot).
It would be great to have similar functionality in clj-kondo (very handy in refactoring).

**Describe the solution you'd like**

```clojure
(ns deprecated-ns
  {:deprecated true})

(defn in-deprecated "Docstring" [])

(ns lint-deprecated-ns
  (:require [deprecated-ns]))

(deprecated-ns/in-deprecated)

```
There are no clj-kondo warnings above (version 2021.06.18).
I would like to see that `deprecated-ns` is deprecated.

**Describe alternatives you've considered**
Cursive plugin for IntellijIDEA.

**Additional context**
![2021-06-29_deprecated-idea-cursive](https://user-images.githubusercontent.com/527113/123769787-eeb81680-d8c9-11eb-9617-ae3ef59b71b8.png)

**Comment by borkdude** [comment]
I think this is a duplicate of #1125

**Comment by borkdude** [comment]
Consider upvoting #1125 with a thumbs up if you want to see it implemented earlier.
